src/main/daml/Daml/Finance/Settlement/Instruction.daml

```diff
@@ -51,13 +48,14 @@ template Instruction
     signatory requestors, signed
     observer step.sender, step.receiver, settlers, Disclosure.flattenObservers observers
 
-    key (requestors, batchId, id) : (Parties, Id, Id)
-    maintainer fst3 key
+    key InstructionKey with requestors; batchId; id : InstructionKey
+    maintainer key.requestors
 
     let
       messageSuffix = " / instruction id = " <> show id
-      discloseA account = Account.disclose (show batchId, settlers) account.owner (S.singleton account.owner) account
-      undiscloseA account = Account.undisclose (show batchId, settlers) account.owner (S.singleton account.owner) account
+      context = show batchId <> "(" <> show id <> ")"
+      discloseA account = Account.disclose (context, settlers) account.owner (S.singleton account.owner) account
```

Comment:
exactly, the settler needs visibility to the contracts it orchestrates during settlement execution.

Reply:
True, it also needs to verify that the account exists in the first place, if I am not mistaken.

The `Debit` is not strictly necessary I think, we would just need to make `Transfer` a consuming choice rather than a nonconsuming one and the `Debit` would happen automatically. Unless this changed with your recent account changes.